controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -258,6 +294,11 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 		return r.upgradeControlPlane(ctx, cluster, kcp, ownedMachines, requireUpgrade)
 	}
 
+	err = r.updateKubeProxyImage(ctx, cluster, kcp)
+	if err != nil{
+		logger.Error(err,"failed updating kube-proxy image")
+	}
```

Comment:
It might be good to add a comment here that we are not waiting for all Nodes in the kubernetes cluster to be updated before updating the kube-proxy daemonset here and only the control plane Machines to match the behavior of kubeadm upgrade.